### DISCUSSION TITLE
Failed to evaluate function: Modelica.Math.Matrices.LAPACK.dhseqr when upgrading to Buildings v10.0.0

### DISCUSSION THREAD

**Discussion by javiarrobas** [lbl-srg/modelica-buildings#3553] [Q&A]
Hi @mwetter and @JayHuLBL 👋🏻
At [this repository](https://gitlab.kuleuven.be/positive-energy-districts/moped) we configure district models using component models from Buildings and IDEASv3.0.0 for a project we are working on. Our testing environment uses Buildingspy with OpenModelica to compile the models and check reference results. The environment is defined in the Dockerfile at [this folder](https://gitlab.kuleuven.be/positive-energy-districts/moped/-/tree/main/MoPED/Resources/Scripts/tests), and the InstallThirdPartyLibraries.mos is used to install a specific version of each library dependency: IDEAS and Buildings. 
I'm trying to upgrade our testing environment from Buildings 9.1.1. to Buildings 10.0.0., but upon doing so I get the following error when compiling some of our models that I'm not able to circumvent. 

```
[/home/developer/.openmodelica/libraries/Modelica 4.0.0+maint.om/Math/package.mo:7653:5-7842:15:writable] Error: External function ‘dhseqr‘ could not be found in any of the given shared libraries:
  /usr/bin/../lib/x86_64-linux-gnu/omc/lapack.so
  /home/developer/.openmodelica/libraries/Modelica 4.0.0+maint.om/Resources/Library/linux64/lapack.so
  /home/developer/.openmodelica/libraries/Modelica 4.0.0+maint.om/Resources/Library/lapack.so
  /usr/bin/../lib/x86_64-linux-gnu/omc/liblapack.so
  /home/developer/.openmodelica/libraries/Modelica 4.0.0+maint.om/Resources/Library/linux64/liblapack.so
  /home/developer/.openmodelica/libraries/Modelica 4.0.0+maint.om/Resources/Library/liblapack.so
  /usr/bin/../lib/x86_64-linux-gnu/omc/ffi/lapack.so
  /usr/bin/../lib/x86_64-linux-gnu/omc/ffi/liblapack.so
[/home/developer/.openmodelica/libraries/Modelica 4.0.0+maint.om/Math/package.mo:10492:9-10492:55:writable] Error: Failed to evaluate function: Modelica.Math.Matrices.LAPACK.dhseqr.
```

See e.g. [here](https://gitlab.kuleuven.be/positive-energy-districts/moped/-/jobs/869121#L231). 

I've tried the following without success:

- Change to older Python versions, in case Lapack was installed with some Python package and the function `dhseqr` was deprecated in the newest versions. 
- Switch between `release`, `stable`, and `nightly` versions of OMC. 
- Install Liblapack from source. 

Notice that I get the same error when using an OMEdit installed in Windows (only when switching to Buildings v9.1.1.):
<img width="646" alt="image" src="https://github.com/lbl-srg/modelica-buildings/assets/22677958/acf8606a-dad2-464b-b582-d91653ffd43f">

My guess is that we never needed `dhseqr` until using Buildings 10.0.0 for our models, but I'm not able to evaluate why or to what extent this is needed because I can't identify where the Buildings library is triggering this function call. Have you experienced some similar issue with your Buildings test suite when using OpenModelica?

If you want to reproduce the issue you can do it just by simulating `MoPED.Districts.Examples.Layout4`. That model works when using Buildings v9.1.1., but crashes with the error above when using Buildings v10.0.0.

Any help or thoughts would be very much appreciated!

----
To complement the information above:

I've tried to isolate the problem from Modelica and seems there is a general issue with that particular `dhseqr` function from Lapack because I can successfully run the following example in Python which uses the lu decomposition from Lapack:

```Python
import numpy as np
from scipy.linalg import lu

# Create a random square matrix
n = 3
A = np.random.rand(n, n)

# Perform LU decomposition using LAPACK
P, L, U = lu(A)

# Check if the LU decomposition was successful
if np.allclose(np.dot(P, np.dot(L, U)), A):
    print("LAPACK LU decomposition successful:")
    print("P (permutation matrix):")
    print(P)
    print("L (lower triangular matrix):")
    print(L)
    print("U (upper triangular matrix):")
    print(U)
else:
    print("LAPACK LU decomposition failed!")
```

But when trying to use `dhseqr` in the following example, I get the error below:
``` Python
import numpy as np
from scipy.linalg import lapack

# Create a random symmetric matrix
n = 3  # Size of the matrix
A = np.random.rand(n, n)
A = (A + A.T) / 2  # Make it symmetric

# Call dhseqr for eigenvalue decomposition
T, Z, info = lapack.dhseqr(A)

if info == 0:
    print("Eigenvalues (real part):")
    print(T)
else:
    print("LAPACK dhseqr failed with info code:", info)
```
```bash
root@49792943d574:/tmp/lapack-3.9.0/build# python3 testlapack.py
Traceback (most recent call last):
  File "/tmp/lapack-3.9.0/build/testlapack.py", line 10, in <module>
    T, Z, info = lapack.dhseqr(A)
AttributeError: module 'scipy.linalg.lapack' has no attribute 'dhseqr'. Did you mean: 'dpteqr'?
```

**Comment by mwetter** [comment]
@JavierArroyoBastida : I have not seen such an error message before. I wonder if the change in library triggers changes in the generated code, which then causes OpenModelica to call this missing function. This looks more like an installation issue with OpenModelica rather than a Modelica library issue.

**Comment by javiarrobas** [comment]
Thanks a lot for your answer @mwetter! Yes, it seems more of an issue with OpenModelica than with the Buildings library, I'll raise an issue on their end as well. Just wanted to make sure this is not familiar to you either.

**Comment by mwetter** [comment]
See https://github.com/OpenModelica/OpenModelica/issues/11314
